apps/trading/components/last-24h-volume/last-24h-volume.tsx

```diff
@@ -1,102 +1,61 @@
+import type { RefObject } from 'react';
+import { useInView } from 'react-intersection-observer';
 import {
   calcCandleVolume,
   marketCandlesProvider,
 } from '@vegaprotocol/market-list';
 import {
   addDecimalsFormatNumber,
-  t,
   useDataProvider,
   useYesterday,
   isNumeric,
 } from '@vegaprotocol/react-helpers';
 import * as Schema from '@vegaprotocol/types';
-import throttle from 'lodash/throttle';
-import { useCallback, useMemo, useRef, useState } from 'react';
-import * as constants from '../constants';
-import { HeaderStat } from '../header';
+import { useMemo } from 'react';
 import type { Candle } from '@vegaprotocol/market-list';
 
 interface Props {
   marketId?: string;
   positionDecimalPlaces?: number;
-  noUpdate?: boolean;
-  isHeader?: boolean;
+  formatDecimals?: number;
+  inViewRoot?: RefObject<Element>;
   initialValue?: string;
 }
 
 export const Last24hVolume = ({
   marketId,
   positionDecimalPlaces,
-  noUpdate = false,
-  isHeader = false,
+  formatDecimals,
+  inViewRoot,
   initialValue,
 }: Props) => {
-  const [candleVolume, setCandleVolume] = useState<string>(initialValue || '');
   const yesterday = useYesterday();
-  // Cache timestamp for yesterday to prevent full unmount of market page when
-  // a rerender occurs
-  const yTimestamp = useMemo(() => {
-    return new Date(yesterday).toISOString();
-  }, [yesterday]);
+  const [ref, inView] = useInView({ root: inViewRoot?.current });
 
   const variables = useMemo(
     () => ({
       marketId: marketId,
       interval: Schema.Interval.INTERVAL_I1H,
-      since: yTimestamp,
+      since: new Date(yesterday).toISOString(),
     }),
-    [marketId, yTimestamp]
+    [marketId, yesterday]
   );
 
-  const throttledSetCandles = useRef(
-    throttle((data: Candle[]) => {
-      noUpdate || setCandleVolume(calcCandleVolume(data) || '');
-    }, constants.DEBOUNCE_UPDATE_TIME)
-  ).current;
-  const update = useCallback(
-    ({ data }: { data: Candle[] | null }) => {
-      if (data) {
-        throttledSetCandles(data);
-      }
-      return true;
-    },
-    [throttledSetCandles]
-  );
-
-  const { error } = useDataProvider<Candle[], Candle>({
+  const { data } = useDataProvider<Candle[], Candle>({
```

Comment:
in such cases we're loosing memoization: `data` will be always a new object and will cause a re-render

Reply:
and `useThrottledDataProvider` shouldn't be use here?